pkg/operator/ceph/config/conditions.go

```diff
@@ -117,50 +104,69 @@ func translatePhasetoState(phase cephv1.ConditionType) cephv1.ClusterState {
 	}
 }
 
-// Updating the status of Progressing, Updating or Upgrading to False once cluster is Ready
-func checkConditionFalse(context *clusterd.Context, namespace, name string) {
-	tempConditionList := []cephv1.ConditionType{cephv1.ConditionUpdating, cephv1.ConditionUpgrading, cephv1.ConditionProgressing}
-	var tempCondition cephv1.ConditionType
-	for _, conditionType := range tempConditionList {
+// MarkProgressConditionsCompleted Updates the status of Progressing, Updating or Upgrading to False once cluster is Ready
+func MarkProgressConditionsCompleted(context *clusterd.Context, namespace, name string) {
+	conditionsToUpdate := []cephv1.ConditionType{cephv1.ConditionUpdating, cephv1.ConditionUpgrading, cephv1.ConditionProgressing}
+	for _, conditionType := range conditionsToUpdate {
 		if conditionMap[conditionType] == v1.ConditionTrue {
-			tempCondition = conditionType
+			reason := ""
+			message := ""
+			if conditionType == cephv1.ConditionUpdating {
+				reason = "UpdateCompleted"
+				message = "Cluster updating is completed"
+			} else if conditionType == cephv1.ConditionUpgrading {
+				reason = "UpgradeCompleted"
```

Comment:
The UpgradeCompleted condition will be set to `false` when the upgrade is completed? Shouldn't it be `true`?

Reply:
It's `Upgrading` condition and it will be set to false. `UpgradeCompleted` is the reason to which why `Upgrading` is set to false.